internal/server/email/smtp.go

```diff
@@ -0,0 +1,238 @@
+package email
+
+import (
+	"bufio"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+)
+
+type Client struct {
+	*tls.Conn
+	reader *bufio.Reader
+	sync.Mutex
+	skipTLSVerify bool
+}
+
+type Message struct {
+	FromName    string
+	FromAddress string
+	ToName      string
+	ToAddress   string
+	Subject     string
+	PlainBody   []byte
+	HTMLBody    []byte
+}
+
+var client = &Client{}
+
+func (c *Client) connect() (err error) {
+	if c.Conn != nil {
+		return nil
+	}
+
+	if SendgridAPIKey == "" {
+		return fmt.Errorf("Sendgrid API key is not set")
+	}
+
+	conn, err := tls.Dial("tcp", SMTPServer, &tls.Config{
+		MinVersion: tls.VersionTLS12,
+		MaxVersion: tls.VersionTLS13,
+		// nolint:gosec
+		InsecureSkipVerify: c.skipTLSVerify, // only used by tests
+	})
+	if err != nil {
+		return err
+	}
+
+	err = conn.SetDeadline(time.Now().Add(30 * time.Second))
+	if err != nil {
+		return fmt.Errorf("set deadline: %w", err)
+	}
+	c.Conn = conn
+
+	defer func() {
+		if err != nil {
+			c.Conn = nil
+		}
+	}()
+
+	c.reader = bufio.NewReader(c.Conn)
```

Comment:
Why not use use "net/smtp" here? I'm worried this code might be fragile.

I think you can do something like:
```
    auth := smtp.PlainAuth("","mail@infrahq.com", "password", host)
    
    client, err := smtp.NewClient(conn, host)
    if err != nil {
        log.Panic(err)
    }

    // Auth
    if err = client.Auth(auth); err != nil {
        log.Panic(err)
    }
```